spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceEntitlementSpecTest.java

```diff
@@ -0,0 +1,771 @@
+        adminClient.owners().createPool(owner.getKey(), Pools.random(product1)
+            .startDate(start)
+            .endDate(start.plusYears(1L)));
+
+        final String consumerUuid = consumer.getUuid();
+        adminClient.owners().listOwnerPools(owner.getKey()).forEach(entry ->
+            consumerClient.consumers().bindPool(consumerUuid, entry.getId(), 1));
+
+        consumer = userClient.consumers().getConsumer(consumerUuid);
+        consumer.getInstalledProducts().forEach(entry -> assertThat(entry)
+            .returns("ALL", ConsumerInstalledProductDTO::getArch)
+            .returns("3.11", ConsumerInstalledProductDTO::getVersion)
+            .returns("green", ConsumerInstalledProductDTO::getStatus)
+            .returns(start, ConsumerInstalledProductDTO::getStartDate)
+            .returns(start.plusYears(1L), ConsumerInstalledProductDTO::getEndDate));
+    }
+
+    @Test
+    public void shouldAllowAConsumerWithoutExistingEntitlementsToDryRunAnAutoAttachOnSLAButNotFilterOnIt() {
+        String serviceLevel1 = StringUtil.random("VIP");
+        String serviceLevel2 = StringUtil.random("Ultra-VIP");
+        ProductDTO product1 = Products.randomEng()
+            .addAttributesItem(ProductAttributes.SupportLevel.withValue(serviceLevel1));
+        product1 = adminClient.ownerProducts().createProductByOwner(owner.getKey(), product1);
+        ProductDTO product2 = Products.randomEng()
+            .addAttributesItem(ProductAttributes.SupportLevel.withValue(serviceLevel2));
+        product2 = adminClient.ownerProducts().createProductByOwner(owner.getKey(), product2);
+        PoolDTO pool1 = adminClient.owners().createPool(owner.getKey(), Pools.random(product1));
+        PoolDTO pool2 = adminClient.owners().createPool(owner.getKey(), Pools.random(product2));
+
+        Set<ConsumerInstalledProductDTO> installed = Set.of(
+            new ConsumerInstalledProductDTO().productId(product1.getId()).productName(product1.getName()),
+            new ConsumerInstalledProductDTO().productId(product2.getId()).productName(product2.getName()));
+
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(adminClient, owner));
+        ConsumerDTO consumer = Consumers.random(owner)
+            .serviceLevel(serviceLevel1)
+            .installedProducts(installed);
+        consumer = userClient.consumers().createConsumer(consumer);
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        consumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+        assertThat(consumer).returns(serviceLevel1, ConsumerDTO::getServiceLevel);
+
+        // dry run against the set service level:
+        // should return both pools because we no longer filter on consumer's sla
+        // (unless the consumer has existing entitlements, and in this case we don't).
+        List<PoolQuantityDTO> pools = adminClient.consumers().dryBind(consumer.getUuid(), null);
```

Comment:
Do we have any guarantee that the pools will be returned in this specific order every time? If not, the assertions following can use `.map` to map the pools collection to the IDs, then `.containsExactlyInAnyOrder` to specify the list of expected pool IDs:

```
assertThat(pools)
  .hasSize(2)
  .map(PoolQuantityDTO::getPool)
  .map(PoolDTO::getId)
  .containsExactlyInAnyOrder(pool1.getId(), pool2.getId());
```

Looks like this kinda thing potentially occurs in many places. I'll omit explicit comments on each one.

Reply:
From the previous tests, I would say that the proper order is expected.
I will make the change to the contains exactly.